app/Builders/Hierarchy/Hierarchy.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Builders\Hierarchy;
+
+class Hierarchy
+{
+    private $id;
+    private $name;
+    private $model;
+    private $parentColumnName = "parent_id";
+
+    public function __construct(String $model)
+    {
+        $this->model = new $model();
+    }
+
+    public function SetParentColumn($parentColumnName)
+    {
+        $this->parentColumnName = $parentColumnName;
+        return $this;
+    }
+
+    public function SetIdNamePair(array $idNameArray)
+    {
+        $this->id = $idNameArray[0];
+        $this->name = $idNameArray[1];
+        return $this;
+    }
+
+    protected function MakeHierarchicalOptions(&$result = [], $parent_id = null, $level = 0)
```

Comment:
```suggestion
    protected function MakeHierarchicalOptions(array &$result = [], int $parent_id = null,int $level = 0): void
```
Type hinting ga etibor berin

Reply:
Togirlandi